src/modules/Sidebar/Sidebar.js

```diff
@@ -191,6 +162,7 @@ class Sidebar extends Component {
       <Ref innerRef={this.handleRef}>
         <ElementType {...rest} className={classes}>
           {childrenUtils.isNil(children) ? content : children}
+          {visible && <EventStack name={'click'} on={this.handleDocumentClick} />}
```

Comment:
30 lines vs 1 line, exciting result ✌️

Reply:
Very cool, I'll give this a proper review before merging.  It looks like a fantastic idea :D